lib/publiccloud/provider.pm

```diff
@@ -354,16 +354,19 @@ sub create_instances {
         }
         # check guestregister conditional, default yes:
         $instance->wait_for_guestregister() if ($args{check_guestregister});
+
         # Performance data: boottime
-        if (!is_openstack && is_ok_url($url)) {
+        continue unless is_openstack;
+
+        if (is_ok_url($url)) {
```

Comment:
IMHO this change will add no improvement to the original code, where this block was already always skipped if it is `openstack`.

Reply:
moreover using `continue` you skip also the `return @vms` below and this create_istance will return no instance, while it is present instead.